packages/alea-frontend/components/HomeworkManager.tsx

```diff
@@ -0,0 +1,354 @@
+  };
+
+  const handleEdit = (homework: HomeworkInfo) => {
+    setHomeworkId(homework.homeworkId);
+    setHomeworkName(homework.homeworkName);
+    const localDate = new Date(homework.homeworkDate);
+    const formattedDate = `${localDate.getFullYear()}-${String(localDate.getMonth() + 1).padStart(
+      2,
+      '0'
+    )}-${String(localDate.getDate()).padStart(2, '0')}`;
+    setHomeworkDate(formattedDate);
+    setArchive(homework.archive);
+    setFilepath(homework.filepath);
+    setView('edit');
+  };
+  const handleDelete = async () => {
+    if (selectedHomeworkId) {
+      try {
+        const response = await deleteHomework(selectedHomeworkId);
+        setMessage(response.message);
+        setOpenSnackbar(true);
+        getHomeworks();
+      } catch (error) {
+        setMessage(error?.response?.data?.message || 'An error occurred');
+        setOpenSnackbar(true);
+      }
+    }
+    setDeleteDialogOpen(false);
+  };
+
+  const confirmDelete = (homeworkId: number) => {
+    setSelectedHomeworkId(homeworkId);
+    setDeleteDialogOpen(true);
+  };
+
+  const resetForm = (isCancelled = true) => {
+    setHomeworkId(null);
+    setHomeworkName('');
+    setHomeworkDate('');
+    setArchive('');
+    setFilepath('');
+    setView('list');
+    if (isCancelled) {
+      if (homeworkId) {
+        setMessage('Homework updation cancelled');
+      } else {
+        setMessage('Homework creation cancelled');
+      }
+    }
+  };
+
+  const handleCloseSnackbar = () => {
+    setOpenSnackbar(false);
+    setMessage('');
+  };
+  const sourcePath = filepath.replace('xhtml', 'tex');
+
+  return (
+    <Box
+      sx={{
+        width: { xs: '100%', sm: '70%' },
+        margin: '0 auto',
+        padding: '0 16px',
+      }}
+    >
+      <Typography variant="h4" gutterBottom>
+        {t.homeworkManagement}{' '}
+      </Typography>
+      <Box
+        sx={{
+          display: 'flex',
+          flexDirection: 'column',
+          alignItems: 'center',
+          mt: 3,
+          px: 3,
+          py: 3,
+          border: '1px solid #ccc',
+          borderRadius: 2,
+        }}
+      >
+        {view === 'list' && (
+          <>
+            <Box
+              sx={{
+                display: 'flex',
+                justifyContent: 'space-between',
+                width: '100%',
+                mb: 2,
+              }}
+            >
+              <Typography variant="h5" gutterBottom sx={{ fontWeight: 700 }}>
+                {t.homeworks}
+              </Typography>
+              <Button
+                variant="contained"
+                color="primary"
+                onClick={() => setView('create')}
+                sx={{
+                  borderRadius: '25px',
+                  marginLeft: '5px',
+                }}
+              >
+                {t.createHomework}{' '}
+              </Button>
+            </Box>
+
+            <TableContainer component={Paper} sx={{ maxHeight: '500px', overflowY: 'auto' }}>
```

Comment:
move this to a separate component. Preferably outside this file. As a rule of thumb, keep files smaller than 200 lines whenever possible.

Reply:
Refactored!